### ISSUE TITLE
Can we use this to uncompress .lz4 files? (URGENT)

### ISSUE THREAD

**Issue by dakshcs** [lz4/lz4#934]
I urgently require a stock android flash on my phone and the stock firmware was compressed in .lz4. I need to uncompress it back to a .img file to mount it, please reply as quick as possible since I require this urgently.
![Screenshot 2020-10-23 190654](https://user-images.githubusercontent.com/73348506/97010383-f10be900-1562-11eb-90c3-7a2c5d96e10d.png)

**Comment by jinfeihan57** [comment]
@phoenix277yt try this : https://github.com/szcnick/p7zip

**Comment by dakshcs** [comment]
> @phoenix277yt try this : https://github.com/szcnick/p7zip

Thanks but it corrupted a csc file...
It's ok for my normal purposes like shader development but my phone is crucial to my daily life so I kinda need a foolproof and reliable method. 
fyi - the firmware is from samsung, normal documents get decompressed flawlessly, its only the firmware files and my phone doesent recognize those, not even odin can help (samsung flash tool)

I tried older firmware from 2019 and February of 2020, it still didnt decompress correctly and managed to corrupt at least 1 file.

I guess its just something with samsung's encoding, but still of you guys find a solution share it with me.

**Comment by jinfeihan57** [comment]
" but it corrupted a csc file... " what do you mean? 
csc file is a  text document， decompression is not required.

**Comment by dakshcs** [comment]
> " but it corrupted a csc file... " what do you mean? 
> csc file is a  text document， decompression is not required.

Yes but its compressed into an img file and then into a lz4 file. When I tried this I used the program you gave me to decompress it and then 7zip to decompress it again to get to the csc file I need to modify. And somewhere in the process there's a problem which corrupts the file. 7zip is working fine so the problem is with samsung encoding or the decompression of lz4.
